perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_virt_eduPersonEntitlement.java

```diff
@@ -0,0 +1,133 @@
+package cz.metacentrum.perun.core.impl.modules.attributes;
+
+import cz.metacentrum.perun.audit.events.AttributesManagerEvents.AllAttributesRemovedForUser;
+import cz.metacentrum.perun.audit.events.AttributesManagerEvents.AttributeRemovedForUser;
+import cz.metacentrum.perun.audit.events.AttributesManagerEvents.AttributeSetForUser;
+import cz.metacentrum.perun.core.api.Attribute;
+import cz.metacentrum.perun.core.api.AttributeDefinition;
+import cz.metacentrum.perun.core.api.AttributesManager;
+import cz.metacentrum.perun.core.api.User;
+import cz.metacentrum.perun.core.api.exceptions.AttributeNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.WrongAttributeAssignmentException;
+import cz.metacentrum.perun.core.impl.PerunSessionImpl;
+import cz.metacentrum.perun.core.implApi.modules.attributes.UserVirtualAttributeCollectedFromUserExtSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * All entitlements collected from:
+ *  - UserExtSources attributes
+ *  - urn:perun:user:attribute-def:virt:groupNames
+ *
+ * @author Dominik Frantisek Bucik <bucik@ics.muni.cz>
+ */
+@SuppressWarnings("unused")
+public class urn_perun_user_attribute_def_virt_eduPersonEntitlement extends UserVirtualAttributeCollectedFromUserExtSource {
+
+	private final Logger log = LoggerFactory.getLogger(this.getClass());
+
+	private final Pattern userAllAttrsRemovedPattern = Pattern.compile("All attributes removed for User:\\[(.*)]", Pattern.DOTALL);
+
+	@Override
+	public String getSourceAttributeFriendlyName() {
+		return "eduPersonEntitlement";
```

Comment:
```suggestion
		return "entitlement";
```
By default SP translate attribute ``eduPersonEntitlement`` to ``entitlement`` and such name must match perun attribute friendly name.

Reply:
For some weird reason when I change the proposed name, the tests start failing. Especially the resolveAttributeValueChangeTest() is problematic. Could you please take a look at it what might be wrong?

Reply:
I'm not sure, tests are working for me when I change it. But I see following auditer message printed in stdout ``[null set for User:[id=<1>, titleBefore=<\0>, firstName=<\0>, lastName=<\0>, middleName=<\0>, titleAfter=<\0>, serviceAccount=<false>, sponsoredAccount=<false>].]`` which is weird, but might be ok for test purpose.